internal/xds/translator/ratelimit.go

```diff
@@ -147,9 +147,26 @@ func buildRouteRateLimits(descriptorPrefix string, global *ir.GlobalRateLimit) [
 			}
 		}
 
+		if rule.IPMatch != nil {
+			// Setup MaskedRemoteAddress action
```

Comment:
from the spec it looks like the masked remote address is being set to RLS https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/route/v3/route_components.proto#config-route-v3-ratelimit-action-maskedremoteaddress
how are individual IPs getting rate limited in a unique way ?

Reply:
this boils down to a top level API question, when a user specifies `192.168.1.0/24`and a limit of 2 requests/hourdoes this mean `192.168.1.1` and `192.168.1.2` are treated as the same client so limit is reached when `192.168.1.1` sends 1 request and `192.168.1.2` sends another request within the same hour ? 
Or
is `192.168.1.1` and `192.168.1.2` allowed 2 requests each ?

Reply:
in that case you should use `192.168.1.1/32` and `192.168.1.1/32`

Reply:
thanks for clarifying , we should clarify this in the API https://github.com/envoyproxy/gateway/blob/27d14f8e9c2babe1a7c9f441114bf59b031b1b6d/api/v1alpha1/ratelimitfilter_types.go#L101 i.e. `All IP Addresses within the specified CIDR are treated as a single client selector and share the rate limit value`

In the future, we could add a `Distinct` match type so each IP is treated in a unique way

Reply:
`MaskedRemoteAddress` is usefull for both Global and Local ratelimit.
In istio, we want to user local ratelimit between pod and pod, pod ip is short live, `marked_remote_address` will be very useful in this scenario.

Reply:
raised https://github.com/envoyproxy/gateway/pull/1145 to clarify docs